spec/ics-027-interchain-account/README.md

```diff
@@ -0,0 +1,292 @@
+
+### Desired Properties  
+
+- Permissionless  
+- Fault containment: Interchain account must follow rules of its host chain, even in times of Byzantine behavior by the counterparty chain (the chain that manages the account)  
+- The chain that controls the account must process the results asynchronously and according to the chain's logic.  
+- Sending and receiving transactions will be processed in an ordered manner.  
+
+## Technical Specification  
+
+The implementation of interchain account is non-symmetric. This means that each chain can have a different way to generate an interchain account and deserialize the transaction bytes and a different set of transactions that they can execute. For example, chains that use cosmos-sdk will deserialize tx bytes by Amino, but if counterparty chain is a smart contract on Ethereum, it may deserialize tx bytes by ABI that is minimal serializing algorithm for smart contract.  
+Interchain account specification defines the general way to register interchain account and transfer tx bytes, and counterparty chain is responsible to deserialize and execute the tx bytes, and sending chain should know how counterparty chain handle tx bytes in advance.
+
+Each chain must satisfy following features to create a interchain account:
+
+- New interchain accounts must not conflict with existing ones.
+- Each chain must keep track of which counterparty chain created each new interchain account.
+
+Also, each chain must know how the counterparty chains serialize/deserialize transaction bytes in order to send transactions via IBC. And the counterparty chain must implement the process of safely exececuting IBC transactions by verifying the authority of the transaction's signers.
+
+The chain must reject the transaction and must not make a state transition in the following cases:
+
+- The IBC transaction fails to be deserialized.
+- The IBC transaction requests for signers that was not made by counterparty chain.
+
+### Data Structures
+
+Each chain must implement below interfaces to support interchain account. ```createOutgoingPacket``` method in ```IBCAccountModule``` interface defines the way how to create an outgoing packet for a specific type. Type indicates how IBC account transaction should be constructed and serialized for the host chain. Generally, type indicates what framework the host chain was built from.  
+```generateAddress``` defines the way how to determine the account's address by using identifier and salt. Using the salt to generate an address is recommended, but not required. If the chain don't support the way to generate an address with salt deterministically, it can be generated by its own way. ```createAccount``` is used to create account with genereated address. New interchain account must not conflict with existing one, and chains should keep track of which counterparty chain created the new interchain account in order to verify the authority of transaction's singers in ```authenticateTx```.
```

Comment:
```suggestion
```generateAddress``` defines the way how to determine the account's address by using identifier and salt. Using the salt to generate an address is recommended, but not required. If the chain doesn't support a deterministic way to generate an address with a salt, it can be generated by its own way. ```createAccount``` is used to create account with generated address. New interchains account must not conflict with existing ones, and chains should keep track of which counterparty chain created each new interchain account in order to verify the authority of transaction's signers in ```authenticateTx```.
```